tasmota/xsns_90_mcp2515.ino

```diff
@@ -0,0 +1,275 @@
+/*
+  xsns_89_mcp2515.ino - MCP2515 CAN bus support for Tasmota
+
+  Copyright (C) 2021 Marius Bezuidenhout
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_SPI
+#ifdef USE_MCP2515
+/*********************************************************************************************\
+ * MCP2515 - Microchip CAN controller
+ *
+ * Connections:
+ * MCP2515  ESP8266         Tasmota
+ * -------  --------------  ----------
+ *  INT     not used
+ *  SCK     GPIO14          SPI CLK
+ *  SI      GPIO13          SPI MOSI
+ *  SO      GPIO12          SPI MISO
+ *  CS      GPIO0..5,15,16  MCP2515
+ *  Gnd     Gnd
+ *  VCC     Vin/5V
+\*********************************************************************************************/
+
+#define XSNS_90              90
+
+#include "mcp2515.h"
+
+// set defaults if not defined
+#ifndef MCP2515_BITRATE
+  #define MCP2515_BITRATE  CAN_500KBPS
+#endif
+
+#ifndef MCP2515_CLOCK
+  #define MCP2515_CLOCK  MCP_8MHZ
+#endif
+
+#ifndef MCP2515_MAX_FRAMES
+  #define MCP2515_MAX_FRAMES 14
+#endif
+
+#ifndef CAN_KEEP_ALIVE_SECS
+  #define CAN_KEEP_ALIVE_SECS 300
+#endif
+
+#ifndef MCP2515_TIMEOUT
+  #define MCP2515_TIMEOUT 10
+#endif
+
+#ifndef MCP2515_BMS_CLIENT
+  #define MCP2515_BMS_CLIENT
+  // Look for Freedom Won BMS data in CAN message
+  #ifndef MCP2515_BMS_FREEDWON
+    #define MCP2515_BMS_FREEDWON
+  #endif  // MCP2515_BMS_FREEDWON
+#endif // MCP2515_BMS_CLIENT
+
```

Comment:
Nice set of defines.

If the user does not define the BMS client it is defined anyway. On the other hand I understand that without any user the current implementation is of no use either.